src/main/java/gregtech/common/metatileentities/multi/multiblockpart/appeng/slot/ExportOnlyAEItemSlot.java

```diff
@@ -0,0 +1,138 @@
+    public void deserializeNBT(NBTTagCompound nbt) {
+        if (nbt.hasKey(CONFIG_TAG)) {
+            this.config = WrappedItemStack.fromNBT(nbt.getCompoundTag(CONFIG_TAG));
+        }
+        if (nbt.hasKey(STOCK_TAG)) {
+            this.stock = WrappedItemStack.fromNBT(nbt.getCompoundTag(STOCK_TAG));
+        }
+    }
+
+    @Override
+    public ExportOnlyAEItemSlot copy() {
+        return new ExportOnlyAEItemSlot(
+                this.config == null ? null : this.config.copy(),
+                this.stock == null ? null : this.stock.copy());
+    }
+
+    @Override
+    public void setStackInSlot(int slot, @NotNull ItemStack stack) {}
+
+    @Override
+    public int getSlots() {
+        return 1;
+    }
+
+    @NotNull
+    @Override
+    public ItemStack getStackInSlot(int slot) {
+        if (slot == 0 && this.stock != null) {
+            return this.stock.getDefinition();
+        }
+        return ItemStack.EMPTY;
+    }
+
+    @NotNull
+    @Override
+    public ItemStack insertItem(int slot, @NotNull ItemStack stack, boolean simulate) {
+        return stack;
+    }
+
+    @NotNull
+    @Override
+    public ItemStack extractItem(int slot, int amount, boolean simulate) {
+        if (slot == 0 && this.stock != null) {
+            int extracted = (int) Math.min(this.stock.getStackSize(), amount);
+            ItemStack result = this.stock.createItemStack();
+            result.setCount(extracted);
+            if (!simulate) {
+                this.stock.decStackSize(extracted);
+                if (this.stock.getStackSize() == 0) {
+                    this.stock = null;
+                }
+            }
+            if (this.trigger != null) {
+                this.trigger.accept(0);
+            }
+            return result;
+        }
+        return ItemStack.EMPTY;
+    }
+
+    @Override
+    public IAEItemStack requestStack() {
+        IAEItemStack result = super.requestStack();
+        if (result instanceof WrappedItemStack) {
+            return ((WrappedItemStack) result).getAEStack();
+        } else {
+            return result;
+        }
+    }
+
+    @Override
+    public IAEItemStack exceedStack() {
+        IAEItemStack result = super.exceedStack();
+        if (result instanceof WrappedItemStack) {
+            return ((WrappedItemStack) result).getAEStack();
+        } else {
+            return result;
+        }
+    }
+
+    @Override
+    public void addStack(IAEItemStack stack) {
+        if (this.stock == null) {
+            this.stock = WrappedItemStack.fromItemStack(stack.createItemStack());
+        } else {
+            this.stock.add(stack);
+        }
+        this.trigger.accept(0);
+    }
+
+    @Override
+    public void setStack(IAEItemStack stack) {
+        if (this.stock == null && stack == null) {
+            return;
+        } else if (stack == null) {
+            this.stock = null;
+        } else {
+            // todo this could maybe be improved with better comparison check
```

Comment:
Is there anything to do more immediately here? Could probably use some more elaboration about what this means.

Reply:
It may be valuable to do a more complex comparison check here, for cases when the stacks are completely identical, no onContentsChanged() call would be needed. But the deep comparison may be more overhead than it saves. I'm not sure (but its also complex to fully compare stacks, so I held off for now)